### PR TITLE
Enable creating reader lists

### PR DESCRIPTION
**Pull request by jsnmoon**
Screen Shot 2020-06-11 at 4 22 39 PM

#### Changes proposed in this Pull Request

* Creates a route for creating a reader list (`/read/list`)
* Enables the user to create a reader list

#### Testing instructions

* Apply D44850-code to public-api.
* Navigate to `/read/list`.
* Enter a list name.
* Press "Save".
* 🎉

### CODE REVIEW

File: client/reader/list-manage/index.jsx [hunk 1]
```diff
@@ -29,8 +29,110 @@ import Main from 'components/main';
...
+					id="list-name"
+					name="list-name"
+					data-key="title"
+					value={ list.title }
+					onChange={ onChange }
+				/>
+				<FormSettingExplanation>The name of the list.</FormSettingExplanation>
+			</FormFieldset>
+
+			<FormFieldset>
```
[the remark this answers is not in the document]
Comment (bluefuton) [reply client/reader/list-manage/index.jsx]: Ah - we still need it for the edit form 👍

**Comment by bluefuton** [comment]
Form text looks good 👍 Can we add `translate()` for all the labels?

**Comment by jsnmoon** [comment]
The API route currently doesn't respect the `is_public` flag; I'll work on spinning up a patch.

File: client/reader/list-manage/index.jsx [hunk 2]
```diff
@@ -27,17 +31,142 @@ import SectionNav from 'components/section-nav';
...
+	const onChange = ( event ) => {
+		const update = { [ event.target.dataset.key ]: event.target.value };
+		if ( 'is_public' in update ) {
+			update.is_public = update.is_public === 'public';
+		}
+		updateList( { ...list, ...update } );
+	};
+	return (
+		<Main>
+			<FormattedHeader headerText="Create List" />
```
Comment (bluefuton) [inline client/reader/list-manage/index.jsx:151]: Need to translate this too.

File: client/reader/list-manage/index.jsx [hunk 3]
```diff
@@ -27,17 +31,142 @@ import SectionNav from 'components/section-nav';
...
+			</FormFieldset>
+
+			<FormFieldset>
+				<FormLabel htmlFor="list-description">{ translate( 'Description' ) }</FormLabel>
+				<FormTextarea
+					data-key="description"
+					id="list-description"
+					name="list-description"
+					onChange={ onChange }
+					placeholder="What's your list about?"
```
Comment (bluefuton) [inline client/reader/list-manage/index.jsx:116]: Need to translate this too.

**Comment by jsnmoon** [comment]
> The API route currently doesn't respect the `is_public` flag; I'll work on spinning up a patch. 

D44850-code.

**Comment by bluefuton** [comment]
Great stuff 👍 Working well for me (aside from the public flag - I'll check out the patch next!)

How about having the route at `/read/list/new` instead?

**Comment by bluefuton** [comment]
One other note that we can address in a different PR - the new list doesn't appear in the sidebar. It is being correctly added to `state.reader.lists.items` though.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3821377

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3821377).

**Pull request merged**